examples/llm/src/models/layers/gpt_blocks.py

```diff
@@ -43,15 +44,17 @@ def __init__(self,
                  alibi: bool = False,
                  resid_pdrop: float = 0.0,
                  low_precision_layernorm: bool = False,
+                 multiquery_attention: bool = False,
                  device: Optional[str] = None,
                  **kwargs):
         del kwargs  # unused, just to capture any extra args from the config
         super().__init__()
 
         layernorm_class = LPLayerNorm if low_precision_layernorm else nn.LayerNorm
+        attn_class = MultiQueryAttention if multiquery_attention else MultiheadAttention
```

Comment:
Do you think we'll stop at 2? Or should we make `attn_class` a choice `str` out of [`multihead`, `multiquery`] the way we do `attn_impl`?

Reply:
Yes, I wanted to do this as part of this PR, but this'll be a next step, wanted to merge this before the foundry migration.